packages/mobile/src/escrow/ReclaimPaymentConfirmationScreen.tsx

```diff
@@ -94,28 +103,41 @@ class ReclaimPaymentConfirmationScreen extends React.Component<Props> {
     return <ReviewHeader title={title} />
   }
 
+  renderFooter = () => {
+    return this.props.isReclaiming ? (
+      <ActivityIndicator size="large" color={colors.celoGreen} />
+    ) : null
+  }
+
   render() {
-    const { t, fee } = this.props
+    const { t, isReclaiming, appConnected } = this.props
     const payment = this.getReclaimPaymentInput()
+    const convertedAmount = divideByWei(payment.amount.toString())
+    const convertedFee = divideByWei(this.getFee().toString())
+
+    const currentBalance = this.props.dollarBalance
+    const userHasEnough = new BigNumber(convertedFee).isLessThanOrEqualTo(currentBalance)
 
     return (
       <View style={styles.container}>
         <DisconnectBanner />
         <ReviewFrame
           HeaderComponent={this.renderHeader}
+          FooterComponent={this.renderFooter}
           confirmButton={{
             action: this.onConfirm,
             text: t('global:confirm'),
-            disabled: false,
+            disabled: isReclaiming || !userHasEnough || !appConnected,
           }}
-          modifyButton={{ action: this.onPressEdit, text: t('cancel'), disabled: false }}
+          modifyButton={{ action: this.onPressEdit, text: t('cancel'), disabled: isReclaiming }}
         >
           <ReclaimPaymentConfirmationCard
-            recipient={payment.recipient}
+            recipientPhone={payment.recipientPhone}
+            recipientContact={payment.recipientContact}
             comment={payment.message}
-            amount={payment.amount}
+            amount={new BigNumber(convertedAmount)}
```

Comment:
I noticed that every place we use divideByWei, we then parse the result into a BIgNumber again. Can you please change divideByWei to return a BigNumber?

Reply:
ooh interesting, will do